sphinx/tutorial/rose/furthertopics/rose-stem.rst

```diff
@@ -0,0 +1,309 @@
+should already be :ref:`familiar with <tutorial-cylc-datetime-utc>`.
+The next part is a Jinja2 block which links the group names the user
+can specify with the :term:`graph <graph>` for that group. In this
+case, the group ``command_spaceship`` gives you the graph:
+
+.. digraph:: Example
+   :align: center
+
+   bgcolor = "none"
+
+   fcm_make_spaceship -> spaceship -> rose_ana_position
+
+This variable ``name_graphs`` is used later to generate the graph when
+the suite is run. The Jinja2 variable ``groups`` is next. This enables you
+to set shortcuts to a list of groups, in this case specifying all on the
+command line will run the tasks associated with both ``command_spaceship``
+and ``fire_lasers``.
+
+The scheduling section contains the Jinja2 code to use the information we
+have already set to generate the graph based on what the user
+requested on the command line.
+
+The runtime section should be familiar. Note, however, that the
+``fcm_make_spaceship`` task sets the environment variable
+``SOURCE_SPACESHIP`` from the Jinja2 variable of the same name. This
+is how the variables passed with ``--source`` on the command line are
+passed to ``fcm-make``, which then uses these environment variables in
+its own configuration files.
+
+
+The ``rose-suite.conf`` file
+----------------------------
+
+The suites associated with ``rose-stem`` require a version number
+indicating the version of the ``rose stem`` command with which they
+are compatible. This is specified in the ``rose-suite.conf`` file,
+together with the default values of ``RUN_NAMES`` and ``SOURCE_SPACESHIP``.
+Paste the following into your ``rose-suite.conf`` file:
+
+.. code-block:: rose
+
+   ROSE_STEM_VERSION=1
+
+   [jinja2:suite.rc]
+   RUN_NAMES=[]
+   SOURCE_SPACESHIP='fcm:spaceship_tr@head'
+
+Both of the Jinja2 variables will be overriden by the user when they
+execute ``rose stem`` on the command line.
+
+
+The ``rose_ana_position`` app
+-----------------------------
+
+The final component is a ``rose ana`` app to test the position of our
+spaceship matches the correct output.
+
+Create an app named ``rose_ana_position`` and paste the following into its
+``rose-app.conf`` file:
+
+.. code-block:: rose
+
+   [ana:grepper.FilePattern(Check X position at each timestep)]
+   pattern='^\s*Position:\s*(.*?)\s*,'
+   files=/home/user/spaceship/kgo.txt
+        =../spaceship/output.txt
+
+   [ana:grepper.FilePattern(Check Y position at each timestep)]
+   pattern='^\s*Position:.*?,\s*(.*?)\s*,'
+   files=/home/user/spaceship/kgo.txt
+        =../spaceship/output.txt
+
+   [ana:grepper.FilePattern(Check Z position at each timestep)]
+   pattern='^\s*Position:.*,\s*(.*)\s*$'
+   files=/home/user/spaceship/kgo.txt
+        =../spaceship/output.txt
+
+This will check that the positions reported by the program match those
+within the known good output file.
+
+
+Known Good Output
+-----------------
+
+In the root of the working copy is a file called ``kgo.txt``.
```

Comment:
I advise moving the sentence on line 225 ('The known good ...') so it directly follows this statement, and maybe change the initial 'the' to 'this', as this seems a more natural grouping and allows the reader to notice that 'kgo' are the initials of known good output, which might otherwise be missed.